### PR TITLE
LPS-138357 Add new view entry logic to boolean field

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://issues.liferay.com/browse/LPS-138357

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @danvb112, can you please check if these failures are related to your changes? Thanks

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/form_instance_records_search_container.jsp [hunk 1]
```diff
@@ -124,22 +124,32 @@ PortletURL portletURL = ddmFormViewFormInstanceRecordsDisplayContext.getPortletU
...
+							for (String value : listValues) {
+								String translatedValue = LanguageUtil.get((Locale)request.getLocale(), value);
+
+								translatedValues.append(translatedValue);
+
+								translatedValues.append(", ");
+							}
+
+							translatedValues.setLength(translatedValues.length() - 2);
+							%>
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/form_instance_records_search_container.jsp:146]: @danvb112, can we move this logic to a dedicated method in the class `ddmFormViewFormInstanceRecordsDisplayContext`?

Moving it to the back-end it's easy to maintain and also allows the possibility to add unit tests, which isn't possible with the logic in JSP. :wink: 

Moreover, Liferay Portal API already provides some util methods that can assist you, please see the code below to get some inspiration:
```
public String getTranslatedColumnValues(String columnValues) {
	String[] translatedColumnValues = Arrays.stream(columnValues.split(","))
		.map(columnValue -> columnValue.trim())
		.map(columnValue -> columnValue.toLowerCase())
		.map(columnValue -> LanguageUtil.get((Locale)request.getLocale(), columnValue))
		.toArray(String[]::new);

	return StringUtil.merge(translatedColumnValues, StringPool.COMMA);
}
```

Comment (danvb112) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/form_instance_records_search_container.jsp]: Thanks for this tip @natocesarrego :smile:

**Comment by danvb112** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @danvb112, can you please check if these failures are related to your changes? Thanks

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordsDisplayContext.java [hunk 2]
```diff
@@ -159,6 +159,30 @@ public int getAvailableLocalesCount() throws Exception {
 		return availableLocales.size();
 	}
 
+	public String getBooleanTranslatedValues(String values, Locale locale) {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordsDisplayContext.java:162]: @danvb112, let's review this pull request together today before or after our daily meeting. :wink:

Comment (danvb112) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordsDisplayContext.java:162]: Ok @natocesarrego, Thanks! :wink:

**Comment by natocesarrego** [comment]
Resent in https://github.com/liferay-forms/liferay-portal/pull/940

**Pull request closed without merging**